deployment/ccip/changeset/cs_ccip_home.go

```diff
@@ -35,29 +36,127 @@ import (
+}
+
 type CCIPOCRParams struct {
 	OCRParameters commontypes.OCRParameters
 	// Note contains pointers to Arb feeds for prices
-	CommitOffChainConfig pluginconfig.CommitOffchainConfig
-	// Note ontains USDC config
-	ExecuteOffChainConfig pluginconfig.ExecuteOffchainConfig
+	CommitOffChainConfig *pluginconfig.CommitOffchainConfig
+	// Note contains USDC config
+	ExecuteOffChainConfig *pluginconfig.ExecuteOffchainConfig
 }
 
-func (c CCIPOCRParams) Validate() error {
+func (c CCIPOCRParams) Validate(selector uint64, feedChainSel uint64, state CCIPOnChainState) error {
 	if err := c.OCRParameters.Validate(); err != nil {
 		return fmt.Errorf("invalid OCR parameters: %w", err)
 	}
-	if err := c.CommitOffChainConfig.Validate(); err != nil {
-		return fmt.Errorf("invalid commit off-chain config: %w", err)
+	if c.CommitOffChainConfig == nil && c.ExecuteOffChainConfig == nil {
+		return errors.New("at least one of CommitOffChainConfig or ExecuteOffChainConfig must be set")
 	}
-	if err := c.ExecuteOffChainConfig.Validate(); err != nil {
-		return fmt.Errorf("invalid execute off-chain config: %w", err)
+	if c.CommitOffChainConfig != nil {
+		if err := c.CommitOffChainConfig.Validate(); err != nil {
+			return fmt.Errorf("invalid commit off-chain config: %w", err)
+		}
+		for tokenAddr, tokenConfig := range c.CommitOffChainConfig.TokenInfo {
+			tokenUnknownAddr, err := ccipocr3.NewUnknownAddressFromHex(string(tokenAddr))
+			if err != nil {
+				return fmt.Errorf("invalid token address %s: %w", tokenAddr, err)
+			}
+			aggregatorUnknownAddr, err := ccipocr3.NewUnknownAddressFromHex(string(tokenConfig.AggregatorAddress))
+			if err != nil {
+				return fmt.Errorf("invalid aggregator address %s: %w", tokenConfig.AggregatorAddress, err)
+			}
+			aggregatorAddr := common.HexToAddress(aggregatorUnknownAddr.String())
+			token := common.HexToAddress(tokenUnknownAddr.String())
+			tokenInfos := make([]tokenInfo, 0)
+			onchainState, ok := state.Chains[selector]
+			if !ok {
+				return fmt.Errorf("chain %d does not exist in state", selector)
+			}
+			for _, tk := range onchainState.BurnMintTokens677 {
+				tokenInfos = append(tokenInfos, tk)
+			}
+			for _, tk := range onchainState.ERC20Tokens {
+				tokenInfos = append(tokenInfos, tk)
+			}
+			for _, tk := range onchainState.ERC677Tokens {
+				tokenInfos = append(tokenInfos, tk)
+			}
+			tokenInfos = append(tokenInfos, onchainState.LinkToken)
+			tokenInfos = append(tokenInfos, onchainState.Weth9)
+			symbol, decimal, err := findTokenInfo(tokenInfos, token)
+			if err != nil {
+				return err
+			}
+			if decimal != tokenConfig.Decimals {
+				return fmt.Errorf("token %s -address %s has %d decimals in provided token config, expected %d",
+					symbol, token.String(), tokenConfig.Decimals, decimal)
+			}
+			feedChainState := state.Chains[feedChainSel]
+			aggregatorInState := feedChainState.USDFeeds[TokenSymbol(symbol)]
+			if aggregatorInState == nil {
```

Comment:
Feels like we should just error if there is no aggregator for a token in the state and we have it in the config as a price that needs to be updated. There is no other price source than the aggregator on the feed chain, as far as I remember.